socs/agents/hwp_supervisor/agent.py

```diff
@@ -750,6 +795,89 @@ def sleep_until_complete(self, session=None, dt=1):
             time.sleep(dt)
 
 
+@contextmanager
+def ensure_grip_safety(hwp_state: HWPState, timeout: float = 60.):
+    """
+    Run required checks for gripper safety. This will check ACU parameters such
+    as az/el position and velocity. If `use_acu_blocking` is set, this will
+    also attempt to block out ACU motion for the duration of the operation.
+
+    Args
+    ------
+    hwp_state : HWPState
+        HWP state object. In addition to reading ACU state vars, this will
+        set the `request_block_ACU_motion` flag if `use_acu_blocking` is set.
+    timeout: float
+        Timeout for waiting for the ACU blockout before an error will be raised.
+    """
```

Comment:
We need to check pid.last_updated is None or not.
Also we need to define max_time_since_update for hwp_state and do the similar check as acu.